test/tools.py

```diff
@@ -0,0 +1,102 @@
+import os
+import tempfile
+
+import vsc.utils.run
+import vsc.mympingpong.tools
+from vsc.install.testing import TestCase
+
+# mock the one imported in the tools module, not run_simple itself
+runs_orig = vsc.mympingpong.tools.run_simple
+_parse_orig = vsc.mympingpong.tools._parse_hwloc_xml
+
+
+class ToolsTest(TestCase):
+    """Test tools"""
+
+    def setUp(self):
+        """restore/set orignal functions mocked in test_hwlocmap"""
+        vsc.mympingpong.tools.run_simple = runs_orig
+        vsc.mympingpong.tools._parse_hwloc_xml = _parse_orig
+        super(ToolsTest, self).setUp()
+
+    def test_hwlocmap(self):
+        """Test hwlocmap"""
+
+        xmlout = "%s/test.xml.%s" % (tempfile.gettempdir(), os.getpid())
+
+        global called
+        called = {'runs': 0, '_parse': 0}
+
+        def runs(cmd):
+            global called
+            called['runs'] += 1
+            print 'runs', called
+            self.assertEqual(cmd, vsc.mympingpong.tools.HWLOC_LS_XML_TEMPLATE % xmlout,
+                             msg='command %s passed to run_simple')
+            return 1, 2 # not relevant, just needs to return 2 things
+
+        def _parse(xml_fn):
+            global called
+            called['_parse'] += 1
+            print '_parse', called
+            return 'randomstring'
+
+        vsc.mympingpong.tools.run_simple = runs
+        vsc.mympingpong.tools._parse_hwloc_xml = _parse
+
+        self.assertEqual('randomstring', vsc.mympingpong.tools.hwlocmap(),
+                         msg='hwlocmap returned output from _parse_hwloc_xml')
+
+        self.assertEqual(called, {'runs': 1, '_parse': 1},
+                         msg='simple_run and _parse both called once by hwlocmap, total %s' % called)
+
+
+    def test_parse_hwloc_xml(self):
```

Comment:
The test will stop after the first assert failure, so you'll have a less complete picture of the failures in this function with a single run of the tests. 
But this is fine for now, I guess you can refactor in to multiple tests when you're debugging issues with this test.